tests/Unit/Core/Context/LegacyControllerContextBuilderTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Core\Context;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\ContextStateManager;
+use PrestaShop\PrestaShop\Core\Context\Employee;
+use PrestaShop\PrestaShop\Core\Context\EmployeeContext;
+use PrestaShop\PrestaShop\Core\Context\LegacyControllerContextBuilder;
+use PrestaShop\PrestaShop\Core\Context\ShopContext;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Tests\Unit\Core\Configuration\MockConfigurationTrait;
+
+class LegacyControllerContextBuilderTest extends TestCase
+{
+    use MockConfigurationTrait;
+
+    public function testBuild(): void
```

Comment:
Are you familiar with the `@dataProvider` in PHPUnit tests? This is more adapted when you mean to test different values with the same test logic You can dynamize the parameters of the test method and pass it multiple settings (from the initialization parameters to the expected values)

It's easier to add multiple use cases then

Reply:
You can find an example here https://github.com/PrestaShop/PrestaShop/blob/f4cea148f09eeda1846f466bbc0a5f10dc277bda/tests/Unit/Core/Context/CurrencyContextBuilderTest.php#L57

This would allow you testing all the special use cases (especially for className and multishop context) efficiently and it reduces the size of code needed while keeping the readability easier (at least in my opinion, but maybe it's because I'm used to it)